modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxState.java

```diff
@@ -180,4 +180,13 @@ public void addActiveCache(GridCacheContext cacheCtx, boolean recovery, IgniteTx
      * @return {@code True} if transaction is empty.
      */
     public boolean empty();
+
+    /**
+     * Takes savepoint from transaction and changes transaction entries to saved state.
+     *
+     * @param savepoint Contain snapshot of entries.
+     * @param cctx Context, where transaction processed.
+     * @param tx Transaction which state must be changed.
+     */
+    public void rollbackToSavepoint(TxSavepoint savepoint, GridCacheSharedContext cctx, IgniteInternalTx tx) throws IgniteCheckedException;
```

Comment:
its better to move method declaration to IgniteTxLocalState, you don't need it somewhere else.

Reply:
IgniteInternalTx  -> GridNearTxLocal, because we cannot rollback other transactions.

Reply:
Agree, I'll do it.